Packs/MicrosoftECM/Integrations/MicrosoftECM/MicrosoftECM.ps1

```diff
@@ -0,0 +1,1074 @@
+
+.PARAMETER date
+The date that should be parsed
+
+.OUTPUTS
+Return The String representation of the datetime object normalized to UTC if or $null if $date is $null
+#>
+Function ParseDateTimeObjectToIso($date)
+{
+	if ($date)
+	{
+		return $date.ToUniversalTime().ToString("yyyy-mm-ddTHH:MM:ssZ")
+	}
+	return $null
+}
+<#
+.DESCRIPTION
+This function Verifies only one of the following arguments was actually given and throws an exception if not.
+
+.PARAMETER $errorMessage
+The error message with which the error should be raised
+
+.PARAMETER $parameters
+The parameters list from which only non-null parameter should be given
+#>
+Function AssertNoMoreThenOneParameterGiven
+{
+	param([Parameter(Mandatory = $true, Position = 0)]
+		[AllowEmptyString()]
+		[string] $errorMessage,
+		[Parameter(Mandatory = $true, Position = 1, ValueFromRemainingArguments = $true)]
+		[AllowEmptyString()]
+		[string[]] $parameters)
+	if (([array]($parameters| where-Object { !!$_ })).Length -gt 1)
+	{
+		throw "Parameter set cannot be resolved using the specified named parameters. $errorMessage"
+	}
+}
+
+<#
+.DESCRIPTION
+This function Verifies none of the following arguments was actually given and throws an exception if it was.
+
+.PARAMETER $errorMessage
+The error message with which the error should be raised
+
+.PARAMETER $parameters
+The parameters list from which all parameters should be null
+#>
+Function AssertNoParameterGiven
+{
+	param([Parameter(Mandatory = $true, Position = 0)]
+		[AllowEmptyString()]
+		[string] $errorMessage,
+		[Parameter(Mandatory = $true, Position = 1, ValueFromRemainingArguments = $true)]
+		[AllowEmptyString()]
+		[string[]] $parameters)
+	if (([array]($parameters| where-Object { !!$_ })).Length -gt 0)
+	{
+		throw "Parameter set cannot be resolved using the specified named parameters. $errorMessage"
+	}
+}
+
+<#
+.DESCRIPTION
+This function Verifies only one of the following arguments was actually given and throws an exception if not.
+For more info see https://docs.microsoft.com/en-us/powershell/module/configurationmanager/get-cmcollection?view=sccm-ps
+
+.PARAMETER collection_id
+Specifies a collection ID
+
+.PARAMETER collection_name
+Specifies a collection name
+
+.PARAMETER distribution_point_group_id
+Specifies the ID of the distribution point group that is associated with the collection
+
+.PARAMETER distribution_point_group_name
+Specifies the name of the distribution point group that is associated with a collection
+
+.OUTPUTS
+Return the used parameter or throws an exception if more then one is used
+#>
+Function ValidateGetCollectionListParams($collection_id, $collection_name, $distribution_point_group_id, $distribution_point_group_name)
+{
+	AssertNoMoreThenOneParameterGiven "Please select only one of: collection_id, collection_name, distribution_point_group_id, distribution_point_group_name."  `
+    $collection_id $collection_name $distribution_point_group_id $distribution_point_group_name
+	$result = ""
```

Comment:
All these types of validations should be done by directly accessing the system var `$args`, it will be more straight forward, simpler, easier to read.
You can tell it's length, it's array so it's easy to check it's contents as well